[NNC] Make NNC sanitize input names

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52614 Avoid dispatch overhead in call to mkldnn convolution
* #52613 Include max pool in fusion groups
* #52513 Extend subgraph utils to cover merging a node following a subgraph
* #52512 Remove unused subgraph vmap api
* #51924 Only functionalize ops which we want to include in mkldnn group
* #51923 Add a filter to remove mutation
* #51922 Dont error on 0-dim in convolution
* #51921 Add 0-dim support for binary mkldnn ops
* #51736 Handle mkldnn broadcasting in mkldnn fuser
* #51600 Add MKLDNN fuser
* **#52786 [NNC] Make NNC sanitize input names**
* #51484 Add support for linear in mkldnn fusion
* #51483 Move conv to mkldnn

Previously, NNC did not sanitize input names. I ran into this in the next PR when making subgraph creation preserve debug names caused a number of NNC cuda failures. I also previously ran into this with some masked_fill failures internally, which led me to disable the operator.

Differential Revision: [D26696699](https://our.internmc.facebook.com/intern/diff/D26696699)

Commit:
Make NNC sanitize input names

[ghstack-poisoned]

Commit:
Update on "[NNC] Make NNC sanitize input names"


Previously, NNC did not sanitize input names. I ran into this in the next PR when making subgraph creation preserve debug names caused a number of NNC cuda failures. I also previously ran into this with some masked_fill failures internally, which led me to disable the operator. 






[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -1702,13 +1702,48 @@ TensorExprKernel::BackendType TensorExprKernel::inferBackendTypeFromDevice(
+  for (const torch::jit::Value* input : graph_->inputs()) {
+    std::string sanitized_name = sanitizeName(input->debugName());
+    // we could get fancier here, but name conflict is extremely unlikely
+    while (name_set.count(sanitized_name)) {
```
We don't really need to make the names unique as TE IR Printer will ensure that (modulo bug #52727 - but that bug doesn't affect correctness, it only affects debuggability). So, we probably could avoid creating the input-name map here.

eellison replied:
I'm going to comment on that issue to remove this logic as part of the task, i still think for debugability id prefer to keep this in until that lands

ZolotukhinM reviewed (approved):
Looks good, I have one just comment (see below).

Commit:
Update on "[NNC] Make NNC sanitize input names"


Previously, NNC did not sanitize input names. I ran into this in the next PR when making subgraph creation preserve debug names caused a number of NNC cuda failures. I also previously ran into this with some masked_fill failures internally, which led me to disable the operator. 






[ghstack-poisoned]

Commit:
Update on "[NNC] Make NNC sanitize input names"


Previously, NNC did not sanitize input names. I ran into this in the next PR when making subgraph creation preserve debug names caused a number of NNC cuda failures. I also previously ran into this with some masked_fill failures internally, which led me to disable the operator.

Differential Revision: [D26696699](https://our.internmc.facebook.com/intern/diff/D26696699)

[ghstack-poisoned]
